### PR TITLE
Improve `IME` documentation

### PR DESCRIPTION
**Pull request by madsmtm**
See #2243.

Please correct me on the specifics here, I may very well be wrong on some of this.

### CODE REVIEW

**Review by kchibisov — APPROVED** [review]

File: src/event.rs [hunk 1]
```diff
@@ -634,7 +637,42 @@ pub struct KeyboardInput {
     pub modifiers: ModifiersState,
 }
 
-/// Describes an event from input method.
+/// Describes [input method](https://en.wikipedia.org/wiki/Input_method) events.
+///
+/// This is also called a "composition event".
+///
+/// Most keypresses using a latin-like keyboard layout simply generate an [`WindowEvent::ReceivedCharacter`].
```
Comment (msiglreith) [inline src/event.rs:644]:
```suggestion
/// Most keypresses using a latin-like keyboard layout simply generate a [`WindowEvent::ReceivedCharacter`].
```

File: src/event.rs [hunk 2]
```diff
@@ -634,7 +637,42 @@ pub struct KeyboardInput {
...
+/// // Press "`" key
+/// Ime::Preedit("`", Some(0), Some(0))
+/// // Press "E" key
+/// Ime::Commit("é")
+/// ```
+///
+/// Additionally, certain input devices are configured to display a candidate box that allow the user to select the
+/// desired character interactively. (To properly position this box, you must use [`Window::set_ime_position`].)
+///
+/// An example of a keyboard layout which uses candiate boxes is pinyin. On a latin keybaord the following event
```
Comment (msiglreith) [inline src/event.rs:660]:
```suggestion
/// An example of a keyboard layout which uses candidate boxes is pinyin. On a latin keybaord the following event
```

File: src/event.rs [hunk 3]
```diff
@@ -634,7 +637,42 @@ pub struct KeyboardInput {
...
+///
+/// An example of a keyboard layout which uses candidate boxes is pinyin. On a latin keybaord the following event
+/// sequence could be obtained:
+/// ```ignore
+/// // Press "A" key
+/// Ime::Preedit("a", Some(1), Some(1))
+/// // Press "B" key
+/// Ime::Preedit("a b", Some(3), Some(3))
+/// // Press left arrow key
+/// Ime::Preedit("a b", Some(1), Some(1))
```
Comment (madsmtm) [inline src/event.rs:668]: I should note: This part is technically not true, currently our macOS implementation gives `Ime::Preedit("a b", Some(3), Some(3))`. But it _should_ give `Ime::Preedit("a b", Some(1), Some(1))`.

Comment (kchibisov) [reply src/event.rs:668]: I can ensure you that it'll work like that on Wayland at least, and that's how it should be on macOS as well? If it doesn't work like that it's likely a macOS bug, though, not really a blocking one...

**Comment by kchibisov** [comment]
@madsmtm I've pushed a commit adding more docs about `Enabled`/`Disabled`.

File: src/event.rs (hunk 3 above)
Comment (madsmtm) [reply src/event.rs:670]: Yeah, it's a bug, should be possible to fix just has to do some UTF-16 to UTF-8 encoding that I didn't really see how to do easily.

**Pull request merged**